File: PropZone/Frontend/PropZone/src/app/models/basic-property-search.model.ts
export interface BasicPropertySearchModel {
  locality?: string;
  city?: string;
  listingPurpose?: 'Sale' | 'Rent'; 
  propertyTypes?: string[]; 
  listerTypes?: string[]; 
  minPrice?: number;
  maxPrice?: number;
  keyword?: string;
  minArea?: number;
  maxArea?: number;
  hasImages?: boolean;
  sortBy?: 'CreatedAt' | 'Price';
  ascending?: boolean;
  status?: 'Available' | 'Sold' | 'Rented';
  isDiscountAvailable?:boolean;

  // admin
  listerId?: string;
  statuses?: string[];

}

File: PropZone/Backend/PropFinderApi/Models/DTOs/Requests/BasicPropertySearchModel.cs
using PropFinderApi.Models.Enums;

namespace PropFinderApi.Models.DTOs
{
    public class BasicPropertySearchModel
    {
        public string? Locality { get; set; }
        public string? City { get; set; }
        public ListingPurpose? ListingPurpose { get; set; }
        public List<ListerType>? ListerTypes { get; set; }
        public List<PropertyType>? PropertyTypes { get; set; }
        public decimal? MinPrice { get; set; }
        public decimal? MaxPrice { get; set; }
        public string? Keyword { get; set; }
        public int? MinArea { get; set; }
        public int? MaxArea { get; set; }
        public bool? HasImages { get; set; }
        public bool? IsDiscountAvailable { get; set; }
    }

}